From e1e7cc024d9a2fc236593d36f2325c5f54fa936a Mon Sep 17 00:00:00 2001
From: "linfang.zhao" <linfang.zhao@amlogic.com>
Date: Fri, 7 Jul 2023 02:40:12 +0000
Subject: [PATCH 03/15] drm: display screen blinking [1/1]

PD#SWPL-128909

Problem:
OSD2 use vpp0

Solution:
OSD2 use vpp1

Verify:
T7C

Test:
DRM-OSD-80

Change-Id: I32951e28d715b32323b831f0d3cb3772f50181e8
Signed-off-by: linfang.zhao <linfang.zhao@amlogic.com>
---
 drivers/drm/vpu-hw/meson_vpu_postblend.c | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/drivers/drm/vpu-hw/meson_vpu_postblend.c b/drivers/drm/vpu-hw/meson_vpu_postblend.c
index ce61083c4..4dd4c1b01 100644
--- a/drivers/drm/vpu-hw/meson_vpu_postblend.c
+++ b/drivers/drm/vpu-hw/meson_vpu_postblend.c
@@ -687,14 +687,12 @@ static void fix_vpu_clk2_default_regs(struct meson_vpu_block *vblk,
 		reg_ops->rdma_write_reg_bits(VPP_VD3_DSC_CTRL, 0x0, 5, 1);
 		reg_ops->rdma_write_reg_bits(MALI_AFBCD_TOP_CTRL, 0x0, 15, 1);
 		reg_ops->rdma_write_reg_bits(MALI_AFBCD_TOP_CTRL, 0x0, 20, 1);
-		/* OSD3  uses VPP1*/
-		osd_set_vpp_path_default(vblk, reg_ops, 3, 1);
-		/* OSD4  uses VPP2*/
-		osd_set_vpp_path_default(vblk, reg_ops, 4, 2);
 	}
 
 	for (i = 0; i < MESON_MAX_OSDS; i++) {
 		if (crtcmask_osd[i] == crtc_index) {
+			/* OSD use the corresponding VPP according to crtcmask_osd */
+			osd_set_vpp_path_default(vblk, reg_ops, i + 1, crtcmask_osd[i]);
 			if (i == 0) {
 				reg_ops->rdma_write_reg_bits(VPP_OSD1_SCALE_CTRL, 0x2, 0, 3);
 			} else if (i == 1) {
-- 
2.25.1

